pool/helpers.go

```diff
@@ -50,6 +56,40 @@ func runJob(ctx context.Context, job amboy.Job, q amboy.Queue, startAt time.Time
 		defer cancel()
 	}
 
+	if err := job.Lock(q.ID()); err != nil {
+		job.AddError(errors.Wrap(err, "problem locking job"))
+		return false
+	}
+	if err := q.Save(ctx, job); err != nil {
+		job.AddError(errors.Wrap(err, "problem saving job state"))
+		return false
+	}
+
+	pingerCtx, stopPing := context.WithCancel(ctx)
+	go func() {
+		iters := 0
+		timer := time.NewTimer(amboy.LockTimeout / 2)
```

Comment:
This would be slightly less code with a ticker, and potentially less error-prone with a ticker. (You don't have to call `Reset`.)